src/Uuid/Uuid.php

```diff
@@ -309,7 +309,8 @@ public function model(bool $lazy = false): Identifiable|null
 				// lazily fill cache by writing to cache
 				// whenever looked up from index to speed
 				// up future lookups of the same UUID
-				$this->populate();
+				// also force to update value again if it is already cached
+				$this->populate($this->isCached());
```

Comment:
We could always pass `true` here. If the UUID is not already cached, the `$force` param in `populate()` doesn't change behavior anyway.

Another thought I had: If the value in the cache was already the same but incorrect, this would write to the cache over and over on every request (resulting in wear on the drive, especially SSDs). This edge case could only happen if there was an incorrect implementation elsewhere (e.g. if `findByCache()` was broken so it would never return a model). If we want to protect against this potential bug, we would need to do this:

- Check if the UUID is already cached.
- If yes, get the value from cache and compare it to the current value.
- If they are the same, throw a `LogicException('Could not find UUID by cache, but the current model ID was already cached before')`.
- Otherwise call `$this->populate(true)` to update the cache with the new value.

Not sure if it's worth it, but it would add extra safety to the implementation.

Reply:
Tbh, I think if we're afraid that `findByCache()` or `findByIndex()` could be incorrectly implemented, we should add more unit tests for these. Not try to mitigate it in other parts if the code.  Those should rely upon `findByCache()` returning the model if its correctly cached as well as `findByIndex()` returning the correct model that holds the uuid.

Reply:
I feel like this is a kind of bug that is really hard to catch in unit tests, but rather easy to catch at runtime.

However I just realized that my proposed fix would be affected by a race condition if the cache is updated by another thread. So it's not going to work.

I'll see if we can build a unit test to verify this edge case.